libraries/rush-lib/src/utilities/PnpmSyncUtilities.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { AlreadyReportedError } from '@rushstack/node-core-library';
+import { Colorize, type ITerminal } from '@rushstack/terminal';
+import {
+  type ILogMessageCallbackOptions,
+  LogMessageIdentifier,
+  type LogMessageDetails,
+  LogMessageKind
+} from 'pnpm-sync-lib';
+
+export function processLogMessage(options: ILogMessageCallbackOptions, terminal: ITerminal): void {
+  const message: string = options.message;
+  const details: LogMessageDetails = options.details;
+
+  // Special formatting for interested messages
+  switch (details.messageIdentifier) {
+    case LogMessageIdentifier.PREPARE_FINISHING:
+      terminal.writeVerboseLine(
+        Colorize.cyan(`pnpm-sync: `) +
+          `Regenerated .pnpm-sync.json in ${Math.round(details.executionTimeInMs)} ms`
+      );
+      return;
+
+    case LogMessageIdentifier.COPY_FINISHING:
+      {
+        const customMessage: string =
+          `Synced ${details.fileCount} ` +
+          (details.fileCount === 1 ? 'file' : 'files') +
+          ` in ${Math.round(details.executionTimeInMs)} ms`;
+
+        terminal.writeVerboseLine(Colorize.cyan(`pnpm-sync: `) + customMessage);
```

Comment:
@g-chao  For some reason this doesn't get printed with `rush rebuild --verbose`.  I didn't have time to investigate why.

Reply:
I checked a little bit, this seems a general bug. Basically, the `--verbose` is not take effect for `rush rebuild` command.

Reply:
The top-level Rush terminal provider is created here:

https://github.com/microsoft/rushstack/blob/e0de053d02bc10128933cb8e75bb1657b636bb3c/libraries/rush-lib/src/cli/RushCommandLineParser.ts#L106-L107

It sets `verboseEnabled` based on the `--debug` flag not the `--verbose` flag:

https://github.com/microsoft/rushstack/blob/e0de053d02bc10128933cb8e75bb1657b636bb3c/libraries/rush-lib/src/cli/RushCommandLineParser.ts#L190-L193

I wonder if this was a historical decision? Maybe because of this definition of `--verbose`?
```
  -v, --verbose         Display the logs during the build, rather than just
                        displaying the build status summary
```
Or maybe because not all commands have the `--verbose` flag?

I'll discuss it with some other maintainers today.

@dmichon-msft

Reply:
In afffe7c7820d5da1a80e0806da3114677a51604f I have wired up the terminal so that `pnpm-sync` output is now printed with `rush --debug`.  To unblock this PR, we'll address the problem where `--verbose` doesn't work in a followup PR.